.circleci/config.yml

```diff
@@ -24,114 +61,281 @@ commands:
+          command: |
+            # aws ecr get-login returns a login command w/ a temp token
+            LOGIN_COMMAND=$(aws ecr get-login --no-include-email --region \
+            $<<parameters.region>> --profile default) # default default default default default default default
+            # save it to an env var & use that env var to login
+            $LOGIN_COMMAND
+      - aws-ecr/build-image:
+          account-url: <<parameters.account-url>>
+          dockerfile: <<parameters.dockerfile>>
+          path: <<parameters.path>>
+          repo: <<parameters.repo>>
+          tag: <<parameters.tag>>
+      - aws-ecr/push-image:
+          account-url: <<parameters.account-url>>
+          repo: <<parameters.repo>>
+          tag: <<parameters.tag>>
 
 jobs:
-  check:
-    docker:
-      - image: circleci/python:3.7
+  terraform-init-and-apply-to-development:
+    executor: docker-terraform
+    steps:
+      - checkout
+      - terraform-init-then-apply:
+          aws-role-name: 'LBH_Circle_CI_Deployment_Role'
+          aws-account: $AWS_ACCOUNT_DEVELOPMENT
+  terraform-init-and-apply-to-staging:
+    executor: docker-terraform
+    steps:
+      - terraform-init-then-apply:
+          aws-role-name: 'LBH_Circle_CI_Deployment_Role'
+          aws-account: $AWS_ACCOUNT_STAGING
+  terraform-init-and-apply-to-production:
+    executor: docker-terraform
+    steps:
+      - terraform-init-then-apply:
+          aws-role-name: 'LBH_Circle_CI_Deployment_Role'
+          aws-account: $AWS_ACCOUNT_PRODUCTION
+  assume-role-development:
+     executor: docker-python
+     steps:
+       - checkout
+       - aws_assume_role/assume_role:
+             account: $AWS_ACCOUNT_DEVELOPMENT
+             profile_name: default
+             role: 'LBH_Circle_CI_Deployment_Role'
+  assume-role-staging:
+     executor: docker-python
+     steps:
+       - checkout
+       - aws_assume_role/assume_role:
+             account: $AWS_ACCOUNT_STAGING
+             profile_name: default
+             role: 'LBH_Circle_CI_Deployment_Role'
+  assume-role-production:
+     executor: docker-python
+     steps:
+       - checkout
+       - aws_assume_role/assume_role:
+             account: $AWS_ACCOUNT_PRODUCTION
+             profile_name: default
+             role: 'LBH_Circle_CI_Deployment_Role'
+  build-and-test:
+    executor: docker-python
     steps:
       - checkout
-      - aws-cli/install
-      - aws-cli/configure:
-          aws-access-key-id: AWS_ACCESS_KEY_ID
-          aws-secret-access-key: AWS_SECRET_ACCESS_KEY
-          aws-region: AWS_REGION
-      - aws-ecr/ecr-login
       - setup_remote_docker
       - run:
           name: build
           command: docker-compose build base-api-test
       - run:
-          name: Run tests
-          command: docker-compose run base-api-test
-
+         name: Run tests
+         command: docker-compose build base-api-test && docker-compose up base-api-test
```

Comment:
One of our devs previously found that when using the docker-compose up command, even if tests fail, CircleCI would not flag it up. However, when you use docker-compose run, it did flag up tests failures.